Sources/Protocols/Designable/GradientDesignable.swift

```diff
@@ -6,145 +6,153 @@
-    insertSubview(gradientView, at: 0)
-  }
+    guard let gradientLayer = makeGradientLayer() else {
+      return
+    }
 
-  func makeGradientLayer() -> CAGradientLayer {
-    let gradientLayer: CAGradientLayer = CAGradientLayer()
-    gradientLayer.frame = bounds
     gradientLayer.cornerRadius = layer.cornerRadius
-    return gradientLayer
+    gradientLayer.frame = bounds
+    let gradientView = PrivateGradientView(frame: bounds, gradientLayer: gradientLayer)
+    insertSubview(gradientView, at: 0)
   }
 
-  func configurePredefinedGradient() -> GradientColor? {
-    guard let gradientType = predefinedGradient else {
-      return nil
-    }
-    return gradientType.colors
-  }
 }
 
-public extension GradientDesignable where Self: UINavigationBar {
-  public func configureGradient() {
-    let predefinedGradient = configurePredefinedGradient()
-    if let startColor = startColor, let endColor = endColor {
-      configureGradient(startColor: startColor, endColor: endColor)
-    } else if let startColor = predefinedGradient?.start, let endColor = predefinedGradient?.end {
-      configureGradient(startColor: startColor, endColor: endColor)
-    }
-  }
+// MARK: - NavigationBar
 
-  fileprivate  func configureGradient(startColor: UIColor, endColor: UIColor) {
+extension GradientDesignable where Self: UINavigationBar {
+
+  public func configureGradient() {
+    let navBarFrame = CGRect(x: 0, y: 0, width: bounds.width, height: UIApplication.shared.statusBarFrame.height)
     let gradientLayer = makeGradientLayer()
-    gradientLayer.colors = [startColor.cgColor, endColor.cgColor]
-    gradientLayer.configurePoints(with: startPoint)
+    gradientLayer?.frame = navBarFrame
+    gradientLayer?.cornerRadius = layer.cornerRadius
 
-    let image = gradientLayer.makeImage()
+    let image = gradientLayer?.makeImage()
     setBackgroundImage(image, for: .default)
   }
 
-  fileprivate func makeGradientLayer() -> CAGradientLayer {
-    let gradientLayer: CAGradientLayer = CAGradientLayer()
+}
 
-    let statusBarHeight: CGFloat = 64 // UIApplication.shared.statusBarFrame.height
-    let navBarFrame = CGRect(x: 0, y: 0, width: bounds.width, height: statusBarHeight)
-    gradientLayer.frame = navBarFrame
+// MARK: - Layers
 
-    gradientLayer.cornerRadius = layer.cornerRadius
-    return gradientLayer
+extension CALayer {
+
+  fileprivate func makeImage() -> UIImage? {
+    UIGraphicsBeginImageContext(frame.size)
+    render(in: UIGraphicsGetCurrentContext()!)
+    let outputImage = UIGraphicsGetImageFromCurrentImageContext()
+    UIGraphicsEndImageContext()
+    return outputImage
   }
 
 }
 
-private class PrivateGradientView: UIView {
+// MARK: - Gradients maker
 
-  init(frame: CGRect, gradientLayer: CAGradientLayer) {
-    super.init(frame: frame)
-    layer.insertSublayer(gradientLayer, at: 0)
-    autoresizingMask = [.flexibleWidth, .flexibleHeight]
+extension GradientDesignable {
```

Comment:
Since all methods are `private`, we can just mark `GradientDesignable` as `private` and remove the others.

Reply:
In fact, that's not working in swift, you can try this in a playground as an example:

```swift
final class Test {

}

private extension Test {

    func test() {
        print("test")
    }

}

let test = Test()
test.test()
```

You can call the method from the private extension from outside even if the extension is private, that's why I moved the private to the methods instead of the extension.

Reply:
@tbaranes interesting, I didn't know that 🤣